### PR TITLE
ci(test): add coverage and split build and run for deterministic test

### PR DESCRIPTION
**Pull request by wangrunji0408**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- add a command `./risedev sslt-cov` to run deterministic e2e with code coverage report.
- run each deterministic test 16 times in CI since we have 16 cores
- ~~add report to buildkite analytics~~
- split the build and run for deterministic test in CI
- add recovery test but allow it to fail

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Comment by huangjw806** [comment]
E2E deterministic test seems that the junit.xml file is not generated.

**Comment by wangrunji0408** [comment]
> E2E deterministic test seems that the junit.xml file is not generated.

It takes time to figure out how to generate junit.xml files. So I decided to postpone it as it's not urgent...

**Review by yezizp2012 — APPROVED** [review]

File: ci/workflows/pull-request.yml [hunk 1]
```diff
@@ -133,15 +145,35 @@ steps:
...
+      # - seek-oss/aws-sm#v2.3.1:
+      #     env:
+      #       BUILDKITE_ANALYTICS_TOKEN: buildkite-build-analytics-deterministic-token
+      - docker-compose#v3.9.0:
+          run: rw-build-env
+          config: ci/docker-compose.yml
+          mount-buildkite-agent: true
+      # - test-collector#v1.0.0:
+      #     files: "*-junit.xml"
+      #     format: "junit"
```
[the remark this answers is not in the document]
Comment (wangrunji0408) [reply ci/workflows/pull-request.yml:173]: It's intended and we'll enable it later.

**Comment by wangrunji0408** [comment]
Seems sqlsmith hunted a bug? #5143

**Pull request merged**